compiler/rustc_resolve/src/macros.rs

```diff
@@ -1131,6 +1131,8 @@ impl<'a> Resolver<'a> {
                 ident.span,
             );
         }
+
+        self.proc_macros = krate.proc_macros.iter().map(|id| self.local_def_id(*id)).collect();
```

Comment:
`compiler\rustc_builtin_macros\src\proc_macro_harness.rs` has direct access to resolver, so it's not necessary to keep the proc macro list in AST.
The harness can collect the list and just write it to resolver (requires adding a method to `ResolverExpand` though).